src/components/custom-field-input-single-choice/custom-field-input-single-choice.jsx

```diff
@@ -34,48 +35,71 @@ export default function CustomFieldInputSingleChoice(props) {
     }
   }
 
-  const listOptions = props.choices.map((item, index) => (
-    <ListOption
-      key={item.id}
-      ref={refs[index]}
-      selected={value && item.id === value.id}
-      value={{
-        id: item.id,
-        label: item.label,
-      }}
-    >
-      {item.label}
-    </ListOption>
-  ));
+  function getOptions() {
+    const choices = {};
+    props.choices.forEach((item, index) => { choices[item.id] = { ...item, index }; });
 
-  function onChange(event) {
+    if (searchValue) {
+      const searchValueLowerCase = searchValue.toLowerCase();
+
+      return props.choices
+        .filter(item => item.label.toLowerCase().includes(searchValueLowerCase))
```

Comment:
At some point, we'll want to ask design / product what kind of filtering algorithm would be ideal. For now, this is good enough.